### PR TITLE
Expect the remote exp to be defined in time zone UTC conform rfc (Fix…

### LINKED ISSUE #1291: Wrong timezone is applied to expiry token
**Describe the bug**
The `_get_token_from_authentication_server` interprets the expire date wrong. The token is read from `content['exp']` which is defined as NumericDate which in turn is defined as UNIX **UTC** Timestamp (seconds since epoch). The `exp` is converted to a date using `utcfromtimestamp` followed by `make_aware` to turn it into a timezone aware date.  The latter uses the timezone as chosen for you project, while it should use the UTC timezone. Now given a timezone +2 and an access token which expires after an hour, the access token will be invalid because it's in the past.

**To Reproduce**
1. Setup identity provider that serves access token with ACCESS_TOKEN_EXPIRE_SECONDS set to 3600
2. Setup a resource  server with timezone set to TIME_ZONE = "Europe/Amsterdam" (+2)
3. Request Access Token and try to access Resource Server
4. Access Token is always expired

**Expected behavior**
Access token is invalid

**Version**
django-oauth-toolkit==2.3.0

- [x] I have tested with the latest published release and it's still a problem.
- [x] I have tested with the master branch and it's still a problem.

**Additional context**
It's rather easy to fix changing:

```
- expires = make_aware(expires) if settings.USE_TZ else expires`
+ expires = make_aware(expires, timezone=pytz.UTC) if settings.USE_TZ else expires`
```

For backwards compatibility it would be better to has a setting which allows you to changes the timezone specifically for the expire token.

### PR DESCRIPTION
**Pull request by wkleinheerenbrink**
Fixes #1291

## Description of the Change
Interpret the EXP in access tokens as being UTC. Introduces an extra setting `AUTHENTICATION_SERVER_EXP_TIME_ZONE` to allow for the workaround in case of a remote authentication server that, wrongly, does not server the EXP in UTC.

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [x] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by dopry** [comment]
@wkleinheerenbrink  could you rebase this PR to resolve the conflicts? I'll prioritize reviewing it in the next few days.

**Comment by makeevolution** [comment]
I have this issue too; can I continue this?

**Comment by n2ygk** [comment]
https://docs.python.org/3/library/zoneinfo.html was added in Python 3.9 so the 3.8 tests fail.

File: setup.cfg [hunk 1]
```diff
@@ -40,6 +40,7 @@ install_requires =
 	requests >= 2.13.0
 	oauthlib >= 3.1.0
 	jwcrypto >= 0.8.0
+	pytz >= 2024.1
```
Comment (n2ygk) [inline setup.cfg:43]: pytz required when Python < 3.9

**Review by n2ygk — APPROVED** [review]
Thanks for this. I've made a few fixes that were causing tests to fail for Python < 3.9

**Pull request merged**